### ISSUE TITLE
database migrate

### ISSUE THREAD

**Issue by cequn** [ibleducation/jupyter-edx-grader-xblock#7]
I am using hawthorn, 
when i follow the instruction in the step database migration, i got this error:
`Traceback (most recent call last):
  File "./manage.py", line 118, in <module>
    startup.run()
  File "/edx/app/edxapp/edx-platform/cms/startup.py", line 19, in run
    django.setup()
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/apps/registry.py", line 85, in populate
    app_config = AppConfig.create(entry)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/apps/config.py", line 94, in create
    module = import_module(entry)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/xblock_jupyter_graded/__init__.py", line 1, in <module>
    from .xblock_jupyter_graded import JupyterGradedXBlock
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/xblock_jupyter_graded/xblock_jupyter_graded.py", line 13, in <module>
    import nbgrader_utils as nbu
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/xblock_jupyter_graded/nbgrader_utils.py", line 11, in <module>
    import container_manager as cm
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/xblock_jupyter_graded/container_manager.py", line 7, in <module>
    from models import Requirement
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/xblock_jupyter_graded/models.py", line 1, in <module>
    from django.contrib.auth.models import User
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/auth/models.py", line 4, in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/auth/base_user.py", line 52, in <module>
    class AbstractBaseUser(models.Model):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/db/models/base.py", line 110, in __new__
    app_config = apps.get_containing_app_config(module)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/apps/registry.py", line 247, in get_containing_app_config
    self.check_apps_ready()
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/apps/registry.py", line 125, in check_apps_ready
    raise AppRegistryNotReady("Apps aren't loaded yet.")
django.core.exceptions.AppRegistryNotReady: Apps aren't loaded yet.
`

**Comment by cequn** [comment]
i also test it on my ginkgo instance, it woks well, 
but one is that the download and upload file seems annoying, can it show the notebook directly to students like the jupyter_viewer? then maybe they can code and submit it

**Comment by labarba** [comment]
What you write is a feature request.

The current version doesn't do what you suggest, as you know. Implementing new features depends on securing new funding for a version 2.

In any case, your feature request is unclear: one thing is _viewing_ the notebook, and quite another is to _interact_ with the notebook, adding code and executing it. The latter requires providing a JupyterHub, and integrating it in Open edX. That is a big development effort, and implies the course provider also supplies cloud compute, which can lead to tremendous cloud bills.

**Comment by roblhibbard** [comment]
same error as above. Does this not work with the Hawthorn version?

**Comment by tu-nv** [comment]
@roblhibbard you can see pull request #8

**Comment by roblhibbard** [comment]
I tried the pull request and now I get errors when trying to install nbgrader's dependents. it is trying to install ipython 6.0+ which is not compatible with the version of python installed with openedx.  Any suggestions.

**Comment by tu-nv** [comment]
@roblhibbard you can pip install version 5+ of ipython manually first.

**Comment by roblhibbard** [comment]
Ok I was able to add manually install and it seems as if Jupyter xblock is installed, but I can’t seem to access it anywhere in the studio. I have added it to the advanced settings for the course. Does it show up under the advanced problem? I am building this off of regis/tutor. 

 

Thank you for you help,

 

Robert Hibbard

**Comment by tu-nv** [comment]
I tested and it works with Open edX docker devstack. which step did you do? can you inspect the studio logs to see whether there are related errors?

**Comment by roblhibbard** [comment]
Hello,

 

I finally got it to work with regisb/tutor docker setup.

 

I had to add both IPython 5.0 and ipykernel 4.10 to the main dockerfile. 

 

Thanks for the tip.

 

Robert

**Comment by roblhibbard** [comment]
I got the jupyter-xblock to show now, but can’t upload instructor notebook, errno 2 no such file or directory.

 

2019-01-29 04:19:07,114 INFO 10 [xblock_jupyter_graded.file_manager] file_manager.py:101 - Found solution cell

cms_1             | 2019-01-29 04:19:07,115 INFO 10 [xblock_jupyter_graded.file_manager] file_manager.py:63 - Wrote file: /var/www/nbgrader/courses/CHS_PY101_2019_S2/source/xblock_jupyter_graded_block_6be46edb8dea46cebf8322216fe3b299/problem3.ipynb

cms_1             | 2019-01-29 04:19:07,127 ERROR 10 [xblock_jupyter_graded.xblock_jupyter_graded] xblock_jupyter_graded.py:266 - [Errno 2] No such file or directory

cms_1             | Traceback (most recent call last):

cms_1             |   File "/usr/local/lib/python2.7/dist-packages/xblock_jupyter_graded/xblock_jupyter_graded.py", line 246, in handle_instructor_nb_upload

cms_1             |     str(self.course_id), str(self.location), f)

cms_1             |   File "/usr/local/lib/python2.7/dist-packages/xblock_jupyter_graded/nbgrader_utils.py", line 54, in generate_student_nb

cms_1             |     max_score = _run_assign_container(f.filename, course, unit)

cms_1             |   File "/usr/local/lib/python2.7/dist-packages/xblock_jupyter_graded/nbgrader_utils.py", line 79, in _run_assign_container

cms_1             |     build_container_if_not_exists(course_id)

cms_1             |   File "/usr/local/lib/python2.7/dist-packages/xblock_jupyter_graded/nbgrader_utils.py", line 211, in build_container_if_not_exists

cms_1             |     if not manager.container_exists():

cms_1             |   File "/usr/local/lib/python2.7/dist-packages/xblock_jupyter_graded/container_manager.py", line 158, in container_exists

cms_1             |     p = Popen(cmd, stdout=PIPE, stderr=PIPE)

cms_1             |   File "/usr/lib/python2.7/subprocess.py", line 711, in __init__

cms_1             |     errread, errwrite)

cms_1             |   File "/usr/lib/python2.7/subprocess.py", line 1343, in _execute_child

cms_1             |     raise child_exception

cms_1             | OSError: [Errno 2] No such file or directory

 

 

any ideas on this one,

 

thank you,

 

Robert

**Comment by geoff-va** [comment]
Check `/var/www/` for a dir called `nbgrader` and see if it's owned by `www-data`. Also make sure that everything is correct in your sudoers file as specified [here](https://github.com/ibleducation/jupyter-edx-grader-xblock#docker-installation).

I just installed the merge request on Hawthorn and I accidentally had it set to `/user/...` instead of `/usr/...` and the directories were created by root instead of www-data, which borked it all up. I deleted the `nbgrader` dir in `/var/www`, fixed the sudoers file and re-uploaded and everything worked fine after that.

I'm going to close this since your question is un related to the original request and I merged @tu-nv 's request (Thanks for that!). Feel free to open a new issue if this persists.

**Comment by roblhibbard** [comment]
I tried this out on Devstack docker, I was able to get it to install, but the problem seems to be running a docker container inside another docker container. Which I was not originally thinking about, but when you go to load an instructor notebook, it can't create the docker container because docker is not installed on the image.

I will probably need to setup a native install and go that route. Unless anyone has an idea on how to run a docker within a docker.

thanks again, just thought I would throw this out there, no need to reopen.

Robert Hibbard
